### ISSUE TITLE
wishlist: BMC updating and config management

### ISSUE THREAD

**Issue by swills** [tinkerbell/rufio#11]
Would be nice to have rufio manage or facilitate BMC updates, and also config changes

## Context
It would be a "nice to have" if rufio supported managing both the version of the BMC software itself as well as updating it. Along with that, being able to use it to push a BMC config would help avoid having to touch that stuff manually remotely. This might include vendor specific things for certain vendors such as Dell or Supermicro. This might also be via Redfish as well. "Enterprise" BMC licenses might be required in some cases.

**Comment by chrisdoherty4** [comment]
Thanks @swills. We'll definitely look at adding this. @joelrebel does BMC lib already facilitate firmware/software updates?

**Comment by joelrebel** [comment]
Yep, bmclib currently supports firmware updates over Redfish - its functional on Dells (and should work on HPEs),
https://github.com/bmc-toolbox/bmclib/blob/master/examples/v1/install-firmware/main.go
We have work planned for Supermicro's. 

The other half of the suggestion above was to have BMC configuration management which is in a similar state,
although I'd say they are two different asks and need to be dealt with separately.

Currently with Redfish, each hardware vendor and generation has different requirements to be met, although Redfish on Dell is pretty much the standard which bmclib tries to cover for.
